pymongo/pool.py

```diff
@@ -1044,6 +1046,35 @@ class _PoolClosedError(PyMongoError):
     pass
 
 
+class _PoolGeneration(object):
+    def __init__(self):
+        # Maps service_id to generation.
+        self._generations = collections.defaultdict(int)
+        # Overall pool generation.
+        self._generations[None] = 0
```

Comment:
Can we use a better sentinel than `None` for tracking overall generation ID? Does this even need to be in the `_generations` map? I think we can just track it as an integer class attribute.

Reply:
LGTM!